File: src/types/lead.ts
export enum LeadStatus {
  NEW = 'NEW',
  CONTACTED = 'CONTACTED',
  QUALIFIED = 'QUALIFIED',
  PROPOSAL = 'PROPOSAL',
  NEGOTIATION = 'NEGOTIATION',
  CONVERTED = 'CONVERTED',
  LOST = 'LOST'
}

export enum LeadSource {
  WEBSITE = 'WEBSITE',
  REFERRAL = 'REFERRAL',
  SOCIAL_MEDIA = 'SOCIAL_MEDIA',
  EMAIL = 'EMAIL',
  PHONE = 'PHONE',
  OTHER = 'OTHER'
}

export interface Lead {
  id?: string;
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
  company?: string;
  source: LeadSource;
  status: LeadStatus;
  notes?: string;
  createdAt?: Date;
  updatedAt?: Date;
}

export interface LeadFormData {
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
  company?: string;
  source: LeadSource;
  status: LeadStatus;
  notes?: string;
}

export interface LeadStats {
    total: number;
    converted: number;
    conversionRate: number;
}

export interface LeadSourceStats extends LeadStats {
    source: LeadSource;
}

export interface LeadTrendData {
    date: string;
    total: number;
    new: number;
    contacted: number;
    qualified: number;
    converted: number;
    lost: number;
}

export interface LeadFilters {
    status?: LeadStatus;
    source?: LeadSource;
    startDate?: string;
    endDate?: string;
    searchTerm?: string;
} 

File: backend/app/models/lead.py
from sqlalchemy import Column, Integer, String, DateTime, Enum as SQLEnum
from sqlalchemy.sql import func
from app.db.base_class import Base
from enum import Enum

class LeadStatus(str, Enum):
    NEW = "new"
    CONTACTED = "contacted"
    QUALIFIED = "qualified"
    CONVERTED = "converted"
    LOST = "lost"

class LeadSource(str, Enum):
    WEBSITE = "website"
    REFERRAL = "referral"
    SOCIAL = "social"
    EMAIL = "email"
    OTHER = "other"

class Lead(Base):
    __tablename__ = "leads"

    id = Column(Integer, primary_key=True, index=True)
    first_name = Column(String(50), nullable=False)
    last_name = Column(String(50), nullable=False)
    email = Column(String(255), unique=True, index=True, nullable=False)
    phone = Column(String(20))
    company = Column(String(100))
    status = Column(SQLEnum(LeadStatus), default=LeadStatus.NEW)
    source = Column(SQLEnum(LeadSource), default=LeadSource.WEBSITE)
    notes = Column(String(1000))
    
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now()) 